### PR TITLE
Update model_fitter.py

### PR DESCRIPTION
**Pull request by BrettBoval**
The limits provided to the solver need to be physically possible. This implementation enforces the constraints that lower_bound, initial_guess, and upper_bound for t0 are strictly ordered.

This PR addresses issue https://trello.com/c/WPATwGKN/336-county-t0-errors

See https://github.com/covid-projections/covid-data-model/actions/runs/186732637 for the snapshot.

### Please Check
- [ ] Are the [JSON schemas](https://github.com/covid-projections/covid-data-model/tree/master/api/schemas) updated?
- [ ] Are the [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.V1.md) updated?
- [ ] Are tests passing?

### CODE REVIEW

**Comment by tashwoods** [comment]
Why is your snapshot failing?

**Comment by BrettBoval** [comment]
> Why is your snapshot failing?

The snapshot worked as expected (see link in the description). Do you mean the unit tests? That's a data ingestion problem with CDS that's being fixed separately.

**Comment by ghop02** [comment]
> > Why is your snapshot failing?
> 
> The snapshot worked as expected (see link in the description). Do you mean the unit tests? That's a data ingestion problem with CDS that's being fixed separately.

If you pull master in, it should work!

File: pyseir/inference/model_fitter.py [hunk 1]
```diff
@@ -240,36 +240,47 @@ def set_inference_parameters(self):
...
-            t0_guess = list(self.times)[idx_enough_cases]
-
             state_fit_result = load_inference_result(fips=self.state_obj.fips)
-            self.fit_params["t0"] = t0_guess
+            T0_LEFT_PAD = 5
+            T0_RIGHT_PAD = 30
+
+            # See if we can do better based on the actual data
+
+            # Let's look at the time when we've got a fifth reported case in a fips
```
Comment (tashwoods) [inline pyseir/inference/model_fitter.py:249]: I don't see any references to requiring the fifth reported case. What am I missing here?

File: pyseir/inference/model_fitter.py [hunk 2]
```diff
@@ -240,36 +240,47 @@ def set_inference_parameters(self):
...
+            if len(idxs_above_threshold) == 0:
+                # Cumulative didn't reach cutoff, use state values
+                self.fit_params["t0"] = state_fit_result["t0"]
+                self.fit_params["limit_t0"] = [
+                    state_fit_result["t0"] - T0_LEFT_PAD,
+                    state_fit_result["t0"] + T0_RIGHT_PAD,
+                ]
+            else:
+                idx_start = idxs_above_threshold[0]
+                self.fit_params["log10_I_initial"] = np.log10(
```
Comment (tashwoods) [inline pyseir/inference/model_fitter.py:263]: How are you initializing this parameter when the state level results are used for initialization?

File: pyseir/inference/model_fitter.py [hunk 3]
```diff
@@ -240,36 +240,47 @@ def set_inference_parameters(self):
...
-            idx_enough_cases = np.argwhere(
-                np.cumsum(self.observed_new_cases) >= OBSERVED_NEW_CASES_GUESS_THRESHOLD
-            )[0][0]
-            initial_cases_guess = np.cumsum(self.observed_new_cases)[idx_enough_cases]
-            t0_guess = list(self.times)[idx_enough_cases]
-
             state_fit_result = load_inference_result(fips=self.state_obj.fips)
-            self.fit_params["t0"] = t0_guess
+            T0_LEFT_PAD = 5
+            T0_RIGHT_PAD = 30
```
Comment (tashwoods) [inline pyseir/inference/model_fitter.py:245]: This is off in the weeds and not blocking, but I think 30 might be too large of a guess. Also, I totally own that I set it to 30 initially.

File: pyseir/inference/model_fitter.py (hunk 1 above)
Comment (BrettBoval) [reply pyseir/inference/model_fitter.py:249]: Fixed. I wanted to make it the 5th value, but thought better to not change multiple things at once.

File: pyseir/inference/model_fitter.py (hunk 2 above)
Comment (BrettBoval) [reply pyseir/inference/model_fitter.py:263]: I don't understand the question. There are class defaults, which can be overwritten by that hand-tuned csv, which can be overwritten in the counties here. This isn't changing any current logic.

File: pyseir/inference/model_fitter.py (hunk 3 above)
Comment (BrettBoval) [reply pyseir/inference/model_fitter.py:245]: Agreed, but I sensitivity analysis is out of scope here. I am just trying to reinstate the 297 counties we weren't going to show model results.

**Pull request merged**